### DISCUSSION TITLE
where/how do I find someone to pay to set this up cFS for me?

### DISCUSSION THREAD

**Discussion by MarshallShoemaker** [nasa/cFS#793] [Q&A]
I have spent many hours trying to load, configure and run
cFS on my MacBook Pro, Apple M1 Processor, UTM virtual box, Ubuntu OS, without success.
I get many dependency errors and have no experience with the Linux command line, seemingly necessary to sort out all the various issues that arise during installation.
 
Is it a hardware pipe dream to run cFS on a Mac with Ubuntu in virtual machine?
If not, where/how do I find someone to pay to set this up cFS for me?

**Comment by ivanperez-keera** [comment]
I have it set up on docker, which you can run on Mac using docker desktop:

https://github.com/space-ros/space-ros/discussions/147

That image installs a lot of dependencies we need for ROS as well, but it's easy to modify to remove the ROS dependencies. In particular, you won't need the steps in the following sections:

- `# Install ROS 2`
- `# Install dependencies of the ROS - cFS bridge`
- `# Clone bridge first. The bridge has both cFS and ROS content we need before we compile either.`
- `# Customize cFS to run the bridge`
  - However, you want to keep these two steps:
    ```
    RUN cp cfe/cmake/Makefile.sample Makefile
    RUN cp -r cfe/cmake/sample_defs sample_defs
    ```
- `# Deploy the sample talker application and adjust the startup scripts.`
- `# This is only needed because docker by default starts in IPv4. This setting is specific to the JAXA bridge.`
- `# Prepare ROS packages`

Feel free to post your image and I think we can help you tweak it until it runs.

**Comment by RichLandau** [comment]
It should work just fine on an Ubuntu VM and may even work on a Mac directly (I've never tried this though). @MarshallShoemaker could you post the logs showing what the errors are?
